routers/api/v1/repo/branch.go

```diff
@@ -122,7 +122,7 @@ func DeleteBranch(ctx *context.APIContext) {
 		return
 	}
 
-	isProtected, err := ctx.Repo.Repository.IsProtectedBranch(branchName, ctx.User)
```

Comment:
If ctx.User == nil should not be removed.

Reply:
Why? It's never called with nil. And even if it would be called the check is not logical there. It does not make a branch protected if the user is nil.

`DeleteBranchPost` redirects to the login page on anonymous acces.
The api `DeleteBranch` results in "user should have a permission to write to a repo".

Here it's covered by line 1450 and 1459:
https://github.com/go-gitea/gitea/blob/77fa7146c6c7f9b1efdc06c24ef7d1f278871e13/routers/repo/issue.go#L1450-L1461
Here it's covered by line 1160:
https://github.com/go-gitea/gitea/blob/77fa7146c6c7f9b1efdc06c24ef7d1f278871e13/routers/repo/pull.go#L1155-L1193

Reply:
It's also enforced by `reqRepoWriter(models.UnitTypeCode)` at:

https://github.com/go-gitea/gitea/blob/77fa7146c6c7f9b1efdc06c24ef7d1f278871e13/routers/api/v1/api.go#L757